internal/common/baseclient/iigo.go

```diff
@@ -52,8 +52,26 @@ func (c *BaseClient) TaxTaken(shared.Resources) {
 
 // GetTaxContribution gives value of how much the island wants to pay in taxes
 func (c *BaseClient) GetTaxContribution() shared.Resources {
-	// TODO: Implement common pool contribution greater than or equal to tax.
-	return 0
+	// If no new communication was received and there is no rule governing taxation, it will still recieve last valid tax
+	gameState := c.ServerReadHandle.GetGameState()
+	presidentCommunications := c.Communications[gameState.PresidentID]
+	var newTaxDecision shared.TaxDecision
+
+	for i := range presidentCommunications {
+		msg := presidentCommunications[len(presidentCommunications)-1-i]
+		if msg[shared.Tax].T == shared.CommunicationTax {
+			newTaxDecision = msg[shared.Tax].TaxDecision
+			break
+		}
+	}
+
+	if newTaxDecision.TaxDecided {
```

Comment:
Can we guarantee that the for loop will always initialise the newTaxDecision? You should really be initialising this to false inacse the for loop does not modify it.